### PR TITLE
Extend filter functionality.

### PR DESCRIPTION
**Pull request by johnharris85**
- Adds multi-select for filters
- Allows permalinks for pre-filtered page loads
- Supports tags with spaces

Signed-off-by: John Harris <joharris@vmware.com>

### CODE REVIEW

File: themes/docsy/assets/js/base.js [hunk 1]
```diff
@@ -27,13 +27,10 @@ limitations under the License.
...
-      var filter = $(this).attr("filter");
-      $(this).addClass("active").siblings().removeClass("active");
-      $(".filter-item").hide();
-      $(".filter-item[class*='" + filter + "']").show();
-      if (filter == "all") {
-        $(".filter-item").show();
-      }
+      if (this.classList.contains("active")) {
+        this.classList.remove("active");
+      } else this.classList.add("active");
```
Comment (zmb3) [inline themes/docsy/assets/js/base.js:32]: John, friends don't let friends use braces for half of an if statement and omit them on the other half :-)

File: themes/docsy/layouts/partials/head-scripts.html [hunk 2]
```diff
@@ -174,4 +174,34 @@
 
   return decodeURI(dc.substring(begin + prefix.length, end));
 }
+
+  function updateFilters() {
+      const activeFilters = Array.from(document.querySelectorAll('li.filter.active')).map(el => el.getAttribute("filter"))
+      window.history.pushState('', '', '?filters=' + activeFilters.join(","));
```
Comment (zmb3) [inline themes/docsy/layouts/partials/head-scripts.html:180]: Does this break if there's already query parameters in the URL?

Maybe we need to use `URLSearchParams` and add a `filters` param:

```
const p = new URLSearchParams(window.location.search)
p.set(filters, activeFilters.join(',')
```

**Comment by BrianMMcClain** [comment]
Only other thing I have to add is do we want to "all" to be an actual filter? I always defer to you all on best practices for this, but intuitively, I would expect the following behavior:

1. Go to a page with filters, no filters in the URL, by default "All" is chosen as it is
2. I choose a filter, "All" is removed and my first chosen filter is the only filter active
3. Any additional filters are added to the list
4. If I chose "All" again, all existing filters are removed (I think?)

Currently, choosing a filter when first entering the page includes "All", so nothing's filtered until I uncheck "all"

Again, I'll look to you two for the best route to go, just my two cents

File: themes/docsy/assets/js/base.js (hunk 1 above)
Comment (johnharris85) [reply themes/docsy/assets/js/base.js]: Fixed.

File: themes/docsy/layouts/partials/head-scripts.html (hunk 2 above)
Comment (johnharris85) [reply themes/docsy/layouts/partials/head-scripts.html]: Fixed (but not like that for "reasons") :p

**Pull request merged**